packages/buidler-etherscan/src/solc/version.ts

```diff
@@ -88,57 +49,49 @@ export async function inferSolcVersion(
     // See https://solidity.readthedocs.io/en/v0.6.0/metadata.html#encoding-of-the-metadata-hash-in-the-bytecode
     if (error instanceof VersionNotFoundError) {
       // The embedded metadata was successfully decoded but there was no solc version in it.
-      const range = {
-        isIncluded(version: SolcVersionNumber): boolean {
-          return semver.satisfies(version.toString(), this.range);
-        },
+      return {
         range: "0.4.7 - 0.5.8",
         inferralType: InferralType.METADATA_PRESENT_VERSION_ABSENT,
-        toString() {
-          return this.range.toString();
-        },
       };
-      return range;
     }
     if (error instanceof MetadataAbsentError) {
       // The decoding failed. Unfortunately, our only option is to assume that this bytecode was emitted by an old version.
-      const range = {
-        isIncluded(version: SolcVersionNumber): boolean {
-          return semver.satisfies(version.toString(), this.range);
-        },
+      return {
         range: "<0.4.7",
         inferralType: InferralType.METADATA_ABSENT,
-        toString() {
-          return this.range.toString();
-        },
       };
-      return range;
     }
     // Should be unreachable.
     throw error;
   }
 
-  return {
-    isIncluded: (version: SolcVersionNumber): boolean => {
-      return (
-        version.major === solcVersionMetadata.major &&
-        version.minor === solcVersionMetadata.minor &&
-        version.patch === solcVersionMetadata.patch
-      );
-    },
+  const range = {
     inferralType: InferralType.EXACT,
-    toString: () => {
-      return `${solcVersionMetadata.major}.${solcVersionMetadata.minor}.${solcVersionMetadata.patch}`;
-    },
+    range: solcVersionMetadata,
   };
+  return range;
+}
+
+// TODO: this could be retrieved from the buidler config instead.
```

Comment:
What do you mean by this?

Reply:
This is a comment to look at the possibility of reusing the `longVersion` retrieved by the internal compiler downloader. Some workflows may avoid the compiler downloader altogether though.